cmd/antrea-agent/agent.go

```diff
@@ -234,11 +236,26 @@ func run(o *Options) error {
 	if features.DefaultFeatureGate.Enabled(features.Traceflow) {
 		go ofClient.StartPacketInHandler(stopCh)
 	}
-	// Create connection store that polls conntrack flows with a given polling interval.
+
+	// Initialize flow exporter to start go routines to poll conntrack flows and export IPFIX flow records
 	if features.DefaultFeatureGate.Enabled(features.FlowExporter) {
-		ctDumper := connections.NewConnTrackDumper(nodeConfig, serviceCIDRNet, connections.NewConnTrackInterfacer())
-		connStore := connections.NewConnectionStore(ctDumper, ifaceStore)
-		go connStore.Run(stopCh)
+		connStore := connections.NewConnectionStore(
+			connections.InitializeConnTrackDumper(nodeConfig, serviceCIDRNet, agentQuerier.GetOVSCtlClient(), o.config.OVSDatapathType),
+			ifaceStore,
+			o.pollInterval)
+		// pollDone helps in synchronizing connStore.Run and flowExporter.Run go routines.
+		pollDone := make(chan struct{})
```

Comment:
you used an unbuffered channel, yet the code in the exporter looks like this:
```
		case <-ticker.C:
			// Waiting for expected number of pollDone signals from go routine(ConnectionStore.Run) is necessary because
			// IPFIX collector computes throughput based on flow records received interval. Number of pollDone
			// signals should be equal to export frequency before starting the export cycle.
			for i := uint(0); i < exp.exportFrequency; i++ {
				<-pollDone
			}
```
I don't understand how that works. Won't the polling code block until the exporter ticker ticks?
As indicated below I don't think the ticker in the exporter is necessary, but the channel should probably be buffered to accommodate `o.config.FlowExportFrequency` signals, *especially* if you keep that ticker.

Reply:
Yes, ticker is not needed anymore. Thanks for pointing it out.
@unbuffered channel: You are right. The initial poll cycles are messed up and behavior is not expected.
I removed the ticker and we can have unbuffered channel as we will not be waiting on bunch of signals simultaneously. Fixed this. Please take a look.